src/core_plugins/input_control_vis/public/components/vis/range_control.js

```diff
@@ -35,94 +34,61 @@ export class RangeControl extends Component {
 
   componentWillReceiveProps = (nextProps) => {
     this.setState(toState(nextProps));
-  }
+  };
 
   handleOnChange = (value) => {
     this.setState({
       sliderValue: value,
       minValue: value.min,
-      isMinValid: true,
+      isRangeValid: true,
       maxValue: value.max,
-      isMaxValid: true,
-      errorMsgs: [],
+      errorMessage: '',
     });
-  }
+  };
 
   handleOnChangeComplete = (value) => {
     this.props.stageFilter(this.props.controlIndex, value);
-  }
+  };
 
   handleMinChange = (evt) => {
-    const min = parseFloat(evt.target.value);
-    const max = this.state.maxValue;
-    if (isNaN(min)) {
-      if (max === '') {
-        this.setState({
-          minValue: '',
-          isMinValid: true,
-          maxValue: '',
-          isMaxValid: true,
-          errorMsgs: [],
-        });
-        return;
-      }
-      this.setState({
-        minValue: '',
-        isMinValid: false,
-        errorMsgs: ['both min and max must be set'],
-      });
-      return;
-
-    } else if (min > max) {
-      this.setState({
-        minValue: min,
-        isMinValid: false,
-        errorMsgs: ['min must be less than max'],
-      });
-      return;
-    }
-
-    this.handleOnChangeComplete({
-      min: min,
-      max: max
-    });
-  }
+    this.handleChange(parseFloat(evt.target.value), this.state.maxValue);
+  };
 
   handleMaxChange = (evt) => {
-    const min = this.state.minValue;
-    const max = parseFloat(evt.target.value);
-    if (isNaN(max)) {
-      if (min === '') {
-        this.setState({
-          minValue: '',
-          isMinValid: true,
-          maxValue: '',
-          isMaxValid: true,
-          errorMsgs: [],
-        });
-        return;
-      }
-      this.setState({
-        maxValue: '',
-        isMaxValid: false,
-        errorMsgs: ['both min and max must be set'],
-      });
-      return;
-
-    } else if (max < min) {
-      this.setState({
-        maxValue: max,
-        isMaxValid: false,
-        errorMsgs: ['max must be greater than min'],
-      });
-      return;
+    this.handleChange(this.state.minValue, parseFloat(evt.target.value));
+  };
+
+  handleChange = (min, max) => {
+    min = isNaN(min) ? '' : min;
+    max = isNaN(max) ? '' : max;
+
+    const isMinValid = min !== '';
+    const isMaxValid = max !== '';
+    let isRangeValid = true;
+    let errorMessage = '';
+
+    if ((!isMinValid && isMaxValid) || (isMinValid && !isMaxValid)) {
+      console.log('both must be set');
```

Comment:
left over console.log statement

Reply:
Good catch. Surprised that passed the linter...